paddle/fluid/framework/details/multi_devices_graph_builder.cc

```diff
@@ -187,15 +188,53 @@ std::unique_ptr<SSAGraph> MultiDevSSAGraphBuilder::Build(
   };
 
   bool is_forwarding = true;
+  int rpc_op_device_id = 0;
+  auto schedule_rpc_op = [&]() -> void {
```

Comment:
Can this be changed to use `get_appropriate_dev` so that when using parallel executor stratagy "Reduce", the variable to send is same as the variable reduced on that device? @chengduoZH am I right?

Reply:
After disscuess with @typhoonzero @chengduoZH @panyx0718 , use `get_appropriate_dev` to schedule rpc op.